doc/developer_guide/psyir.rst

```diff
@@ -168,6 +168,37 @@ Iteration construct
 .. autoclass:: psyclone.psyGen.Loop
     :members:
 
+Ranges
+======
+
+The PSyIR has the `Range` node which represents a range of integer
+values with associated start, stop and step properties. e.g. the list
+of values [4, 6, 8, 10] would be represented by a `Range` with a start
+value of 4, a stop value of 10 and a step of 2 (all stored as `Literal`
+nodes).
```

Comment:
Say that this could be used for loop bounds and array slices but is not at the moment?

Reply:
Yes, done.